pkg/destroy/aws/aws.go

```diff
@@ -289,12 +294,14 @@ func (o *ClusterUninstaller) findResourcesToDelete(
 		}
 	}
 
-	// Find IAM roles
-	iamRoleResources, err := findIAMRoles(ctx, iamRoleSearch, deleted, o.Logger)
-	if err != nil {
-		errs = append(errs, err)
+	// Find IAM roles if this is not an STS cluster
+	if o.CredentialsMode != string(types.ManualCredentialsMode) {
+		iamRoleResources, err := findIAMRoles(ctx, iamRoleSearch, deleted, o.Logger)
+		if err != nil {
```

Comment:
Instead of skipping the finding of IAM roles entirely, we could also check the error for `AccessDenied` and ignore it in that case, either here or inside `findIAMRoles` since it's the `GetRoleWithContext` call that is failing.

Reply:
Also, the installer can still create IAM roles for the control plane nodes in manual mode: https://github.com/openshift/installer/blob/master/data/data/aws/cluster/master/main.tf#L18

Manual mode specifically disables IAM role creation by the cloud credential operator. So if we gate based on manual mode we will be leaking the IAM roles created by the installer.